src/js/modules/articles/reducer.js

```diff
@@ -2,123 +2,83 @@ import {
-      volume: 108,
-      issue: 1,
-      date: "July 19, 2017",
-      time: "8:27 AM",
-      sectionSlug: "creative-writing",
-    },
-  },
+  articles: {},
 };
 
 const reducer = (state = { ...initialState }, action) => {
   switch (action.type) {
     case FETCH_ARTICLE_PENDING: {
-      return { ...state, isFetching: true };
-    }
-    case FETCH_ARTICLE_REJECTED: {
-      return { ...state, isFetching: false, error: action.payload };
+      return { ...state, isFetching: true, };
     }
     case FETCH_ARTICLE_FULFILLED: {
       return {
         ...state,
         isFetching: false,
         isFetched: true,
-        request: action.payload,
+        // action.payload already tested in actions to be array
+        articles: action.payload.reduce((acc, cur) => {
+          const articleSlug = cur.slug;
+          cur.dateline = formatDate(cur.created_at);
+          cur.sectionSlug = slugFinder(cur.section_id, action);
+          delete cur.section_id;
+          acc[ articleSlug ] = cur;
+          return acc;
+        }, { ...state.articles }),
+      };
+    }
+    case FETCH_ARTICLE_REJECTED: {
+      return {
+        ...state,
+        isFetching: false,
+        error: action.payload,
       };
     }
   }
   return state;
 };
 
-export default reducer;
+const formatDate = (createdAt) => {
+  const articleDateline = new Date(createdAt);
+  const dateOffset = Number(articleDateline.toString().slice(28, 31)) * (-1);
+  articleDateline.setHours(articleDateline.getHours() + dateOffset);
+  //Converts the date object into a string to be used for slicing
+  const adjustedArticleDateline = articleDateline.toString();
+  const currentDate = (new Date()).toString().slice(4, 15);
+  if (currentDate === adjustedArticleDateline.slice(4, 15)) {
+    const options = {
+      weekday: "long", year: "numeric", month: "short",
+      day: "numeric", hour: "2-digit", minute: "2-digit"
+    };
+    // articleDateTime is a string in the format <Wednesday, Aug 30, 2017, 8:03 PM>
+    const articleDateTime = articleDateline.toLocaleDateString("en-us", options);
+    return articleDateTime.slice(articleDateTime.indexOf(' ', 20));
+  } else {
+    const months = [ "January", "February", "March", "April", "May", "June",
+      "July", "August", "September", "October", "November", "December" ];
+    return months[ articleDateline.getMonth() ] + ' ' +
+      articleDateline.getDate() + ',' +
+      adjustedArticleDateline.slice(10, 15);
+  }
+};
+
+const slugFinder = (sectionId, action) => {
+  const sections = action.sections;
+  for (sectionIndex in sections) {
+    const sectionObject = sections[ sectionIndex ];
+    if (sectionObject.id === sectionId) {
+      return sectionObject.slug;
+    }
+  }
+  console.log("Section ID of requested article doesn't match")
+};
+
+

```

Comment:
A lot of this shouldn't be necessary. Look further into the documentation. For example, you absolutely do not need a months array. And there is definitely a way to change timezone instead of having a `dateOffset`. In short, much of this code is largely unnecessary and just requires further research into ISO string formatting. <a href="https://docs.microsoft.com/en-us/scripting/javascript/date-and-time-strings-javascript">Take a look</a>.